support/k8s/k8s-workload-registrar/config.go

```diff
@@ -45,33 +52,101 @@ func ParseConfig(hclConfig string) (*Config, error) {
 		return nil, errs.New("unable to decode configuration: %v", err)
 	}
 
-	if c.LogLevel == "" {
-		c.LogLevel = defaultLogLevel
-	}
-	if c.Addr == "" {
-		c.Addr = defaultAddr
+	if c.Mode == "" {
+		c.Mode = "admission"
 	}
-	if c.CertPath == "" {
-		c.CertPath = defaultCertPath
+
+	if err := c.validate(); err != nil {
+		return nil, err
 	}
-	if c.CaCertPath == "" {
-		c.CaCertPath = defaultCaCertPath
+
+	c.setDefault()
+
+	return c, nil
+}
+
+func (c *Config) validate() error {
+	var errGroup errs.Group
+
+	switch c.Mode {
+	case "admission":
+	case "informer":
+		// Note that time.ParseDuration does not accept "" as zero
+		if c.InformerResyncInterval != "" {
+			if _, err := time.ParseDuration(c.InformerResyncInterval); err != nil {
+				errGroup.Add(errs.New("invalid informer_resync_interval %s: %v", c.InformerResyncInterval, err))
+			}
+		}
+	default:
+		errGroup.Add(errs.New("invalid mode %s", c.Mode))
 	}
-	if c.KeyPath == "" {
-		c.KeyPath = defaultKeyPath
+
+	// Validate the 'mode' tag on the struct: can only be set in this mode
+	v := reflect.Indirect(reflect.ValueOf(c))
```

Comment:
I'm not sure I'd hold up the PR for this but this validation could be tighter... e.g, it won't prevent config values being set that happen to populate the field with the zero value.

Reply:
I had a bit of a dig through the HCL documentation and it looks like this would require inspecting the raw AST, which is quite involved. I'm going to punt on that for now.